app/models/job_execution.rb

```diff
@@ -146,24 +154,22 @@ def setup!(dir)
   end
 
   def commands(dir)
-    commands = [
-      "export DEPLOY_URL=#{@job.full_url.shellescape}",
-      "export DEPLOYER=#{@job.user.email.shellescape}",
-      "export DEPLOYER_EMAIL=#{@job.user.email.shellescape}",
-      "export DEPLOYER_NAME=#{@job.user.name.shellescape}",
-      "export REVISION=#{@reference.shellescape}",
-      "export TAG=#{(@job.tag || @job.commit).to_s.shellescape}",
-      "export CACHE_DIR=#{artifact_cache_dir}",
-      "cd #{dir}",
-      *@job.commands
-    ]
-
-    if @stage
-      group_names = @stage.deploy_groups.pluck(:env_value).sort.join(" ")
-      commands.unshift("export DEPLOY_GROUPS=#{group_names.shellescape}") if group_names.present?
-      commands.unshift("export STAGE=#{@stage.permalink.shellescape}")
+    env = {
+      DEPLOY_URL: @job.full_url,
+      DEPLOYER: @job.user.email,
+      DEPLOYER_EMAIL: @job.user.email,
+      DEPLOYER_NAME: @job.user.name,
+      REVISION: @reference,
+      TAG: (@job.tag || @job.commit).to_s,
+      CACHE_DIR: artifact_cache_dir
+    }.merge(@env)
```

Comment:
Oh, right.  I'm thinking of `{ DEPLOY_URL => @job.full_url }`.  That explains why it works.

I don't think I've ever seen uppercase symbols in Ruby, so I was thrown off by that.  I think it sacrifices understandability a bit, so I'd prefer the construct I suggested.  But not that huge a deal.